### PR TITLE
Issue 11211 create directive for ripple effects in buttons and or links

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/app/api/util/ColorUtil.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {Injectable} from '@angular/core';
+
+@Injectable()
+export class ColorUtil {
+    /* Check color brightness
```
Comment (fmontes) [inline src/app/api/util/ColorUtil.ts]: Add tsdoc comments

File: src/app/api/util/ColorUtil.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        let c_r = parseInt(hexCode.substr(0, 2), 16);
+        let c_g = parseInt(hexCode.substr(2, 2), 16);
+        let c_b = parseInt(hexCode.substr(4, 2), 16);
+
+        return ((c_r * 299) + (c_g * 587) + (c_b * 114)) / 1000;
+    }
+
+    /* convert RGB to hex
```
Comment (fmontes) [inline src/app/api/util/ColorUtil.ts]: Add tsdoc comments

File: src/app/api/util/ColorUtil.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * http://stackoverflow.com/questions/1740700/get-hex-value-rather-than-rgb-value-using-jquery
+     */
+    public rgb2hex(rgb): string {
```
Comment (fmontes) [inline src/app/api/util/ColorUtil.ts]: Add tsdoc comments

File: src/app/app-component.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ import {NotLicensedService} from './api/services/not-licensed-service';
     providers: [],
     selector: 'app',
     // TODO: make primeng into main.scss
-    styleUrls: ['app.scss'],
+    styleUrls: ['app.scss', './view/directives/ripple/dot-ripple-effects.scss'],
```
Comment (fmontes) [inline src/app/app-component.ts]: Import this in ```/styles/main.scss```

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import {
```
Comment (fmontes) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: Add tsdoc comments

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            this.renderer.detachView([divElement]);
+        }, this.ANIMATION_DURATION_MILLIS);
+    }
+
+    private createDiv(event: any): any {
+        let hostNativeElement: any = this.host.nativeElement;
+        let divElement = this.renderer.createElement(hostNativeElement, 'div');
+
+        let btnOffset = hostNativeElement.getBoundingClientRect();
+        let  xPos = event.pageX - btnOffset.left;
```
Comment (fmontes) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: extra space.

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        let  yPos = event.pageY - btnOffset.top;
+        let hostHeight = hostNativeElement.clientHeight;
+
+        let divSize = this.getDivSize();
+        let divLeft = (xPos - (divSize.width / 2));
+        let divTop = (yPos - (divSize.height / 2));
+
+        this.renderer.setElementClass(divElement, 'ripple-effect', true);
+        this.renderer.setElementStyle(divElement, 'position',  'absolute');
```
Comment (fmontes) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: Move no dynamic properties to the scss file.

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+            width: rippleSize
+        };
+    }
+
+    private getRippleEffectsColor(): string {
+        let hostNativeElement: any = this.host.nativeElement;
+        let hostBackgroundColor = window.getComputedStyle(hostNativeElement, null).backgroundColor;
+
+        let isBright = this.colorUtil.getBrightness(hostBackgroundColor);
+        return  !isBright ?  this.EFFECT_DEFAULT_COLOR : this.WHITE_COLOR;
```
Comment (fmontes) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: Check extra spaces.

File: src/app/view/directives/ripple/dot-ripple-effects.scss [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    to {
+        opacity: 0;
+        transform: scale(1)
+    }
+}
+
+.ripple-effect {
+    position: 'absolute';
+    border-radius: '50%';
+
```
Comment (fmontes) [inline src/app/view/directives/ripple/dot-ripple-effects.scss]: extra line

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+@Directive({
+    host: {
+        '[style.overflow]': '"hidden"',
+        '[style.position]': '"relative"',
+    },
+    selector: '[dotRipple]'
+})
+export class DotRippleEffectDirective {
+
+    private WHITE_COLOR = 'rgb(255, 255, 255)';
```
Comment (jdotcms) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: looks like a constants, you may use const instead

File: src/app/view/directives/ripple/dot-ripple-effect.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        this.renderer.setElementStyle(divElement, 'width',  `${divSize.width}px`);
+        this.renderer.setElementStyle(divElement, 'height',  `${divSize.height}px`);
+        this.renderer.setElementStyle(divElement, 'background', this.getRippleEffectsColor());
+        this.renderer.setElementStyle(divElement, 'top', `${divTop}px`);
+        this.renderer.setElementStyle(divElement, 'left',  `${divLeft}px`);
+
+        return divElement;
+    }
+
+    private getDivSize(): {width: number, height: number} {
```
Comment (jdotcms) [inline src/app/view/directives/ripple/dot-ripple-effect.ts]: may return an interface!

**Pull request merged**